src/MatrixTwitter.js

```diff
@@ -0,0 +1,1040 @@
+ * @param  {string} userid The Matrix user_id of the timeline owner. This would
+ *                         be a bridge user.
+ * @param  {external:Entry} entry The entry for a room in the RoomBridgeStore.
+ */
+MatrixTwitter.prototype.add_timeline = function (userid, entry) {
+  var obj = {
+    "user_id": userid,
+    "entry": entry
+  };
+  this.timeline_queue.push(obj);
+  log.info('Twitter', "Added Timeline: %s", userid);
+}
+
+
+/**
+ * MatrixTwitter.prototype.remove_timeline - Remove a timeline from being
+ * processed.
+ *
+ * @param  {type} userid The Matrix user_id of the timeline owner. This would
+ *                       be a bridge user.
+ */
+MatrixTwitter.prototype.remove_timeline = function (userid) {
+  const tlfind = (tline) => { tline.user_id == userid };
+  this.timeline_queue = this.timeline_queue.splice(this.timeline_queue.findIndex(tlfind), 1);
+}
+
+
+/**
+ * MatrixTwitter.prototype.stop_timeline - Stop the timeline timer so that
+ * timelines are no longer processed.
+ */
+MatrixTwitter.prototype.stop_timeline = function () {
+  if (this.timeline_intervalID) {
+    clearInterval(this.timeline_intervalID);
+    this.timeline_intervalID = null;
+  }
+}
+
+
+/**
+ * MatrixTwitter.prototype.start_timeline - Start the timeline timer so that
+ * timelines will be processed in turn.
+ */
+MatrixTwitter.prototype.start_timeline = function () {
+  this.timeline_intervalID = setInterval(() => {this._process_timeline();}, TIMELINE_POLL_INTERVAL);
+}
+
+/*
+  Hashtag functions
+*/
+
+
+/**
+ * MatrixTwitter.prototype.start_hashtag - Start the hashtag timer so that
+ * hashtags are processed in turn.
+ */
+MatrixTwitter.prototype.start_hashtag = function () {
+  this.hashtag_intervalID = setInterval(() => {this._process_hashtag_feed();}, HASHTAG_POLL_INTERVAL);
+}
+
+/**
+ * MatrixTwitter.prototype.stop_hashtag - Stop the hashtag timer so that
+ * hashtags are no longer processed.
+ */
+MatrixTwitter.prototype.stop_hashtag = function () {
+  if (this.hashtag_intervalID) {
+    clearInterval(this.hashtag_intervalID);
+    this.hashtag_intervalID = null;
+  }
+}
+
+
+/**
+ * MatrixTwitter.prototype.add_hashtag_feed - Add a hashtag to be processed.
+ *
+ * @param  {string} hashtag The hashtag to add (without the #).
+ * @param  {external:Entry} entry The entry for a room in the RoomBridgeStore.
+ */
+MatrixTwitter.prototype.add_hashtag_feed = function (hashtag, entry) {
+  var obj = {
+    "hashtag": hashtag,
+    "entry": entry
+  };
+  this.hashtag_queue.push(obj);
+  log.info('Twitter', "Added Hashtag Feed: %s", hashtag);
+}
+
+
+/**
+ * MatrixTwitter.prototype.remove_hashtag_feed - Remove a hashtag so that it
+ * will no longer be processed.
+ *
+ * @param  {string} hashtag The hashtag to remove (without the #).
+ */
+MatrixTwitter.prototype.remove_hashtag_feed = function (hashtag) {
+  const htfind = (feed) => { feed.hashtag == hashtag };
+  this.hashtag_queue = this.hashtag_queue.splice(this.hashtag_queue.findIndex(htfind), 1);
```

Comment:
Lovely subtle bug here. What if the hashtag is not found? `htfind` returns `false` for everything, making `findIndex` return `-1` by default, and `Array.splice(-1, 1)` pops the last element of the array.